acstools/acsphotcte.py

```diff
@@ -0,0 +1,289 @@
+            # Iterable in Python.
+            if isinstance(inputs[key], Iterable) and \
+                    not isinstance(inputs[key], list) and \
+                    not isinstance(inputs[key], str):
+                # Convert the array to a python list
+                inputs[key] = list(inputs[key])
+
+                # Check if the first element of the input array is a float
+                if not isinstance(inputs[key][0], float):
+                    LOG.info('Converting elements..')
+                # Convert the array of elements to floats
+                    inputs[key] = list(map(float, inputs[key]))
+
+            # Check if the input is a python float, not an np.float32
+            # Handles the case of str inputs as well
+            elif not isinstance(inputs[key], float):
+                try:
+                    inputs[key] = float(inputs[key])
+                except TypeError as e:
+                    # TypeError during the conversion, print the info and
+                    # return the key it failed on
+                    LOG.info(e)
+                    return key
+                else:
+                    if key == 'radius' and inputs[key] not in [3., 5.]:
+                        LOG.info('Submitted radii not supported...')
+                        return key
+        return inputs
+
+    def _compute_corrections(
+            self,
+            radius=None,
+            ytransfers=None,
+            mjd=None,
+            local_skys=None,
+            fluxes=None,
+            api_id=None,
+            api_key=None,
+    ):
+        """Compute the photometric CTE corrections
+
+        Parameters
+        ----------
+        radius : {3, 5}
+            Aperture radius used to extract photometry
+
+        ytransfers : float, list
+            Number of y transfers. For chip 2 on WFC this is just the Y value,
+            for chip 1 it is 2048 - Y value.
+
+        mjd : float
+            MJD date of the observation
+
+        local_sky : float, list
+            Local sky background for each source
+
+        fluxes : float, list
+            Computed source flux, in electrons, for 3 or 5 pixel apertures.
+
+        Returns
+        -------
+
+        """
+
+        api_credentials = {
+            'x-api-id': api_id,
+            'x-api-key': api_key
+        }
+        inputs = {
+            'radius': radius,
+            'ytransfers': ytransfers,
+            'mjd': mjd,
+            'local_skys': local_skys,
+            'fluxes': fluxes
+        }
+        inputs = self._check_inputs(**inputs)
+        if not isinstance(inputs, dict):
+            LOG.info(f'Did you forget to supply the {inputs}?')
+            return
+        content = self._query(inputs, api_credentials)
+
+        # Update the attribute and return the value
+        self.cte_corrections = np.array(content['deltamag'])
+        return self.cte_corrections
+
+    def correct_photometry(
```

Comment:
Not all the inputs are explained in docstring. Why?

Reply:
The `api_key` and `api_id` arguments are only to be used for development purposes. Is there a proper way to handle this other than just documenting that they're to be used for that?

Reply:
At the very least, state that they are for development purpose only? Although, API doc is usually geared for developers to read. For instance, what do you want the next maintainer to know about these options?